File: 0645-set-mismatch/0645-set-mismatch.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        nums_set = {i for i in range(1, len(nums)+1)}

        res = []
        for i in range(len(nums)):
            if nums[i] not in nums_set:
                res.append(nums[i])
            else:
                nums_set.remove(nums[i])
        
        res.append(nums_set.pop())
        return res

File: 0645-set-mismatch/0645-set-mismatch.go
func findErrorNums(nums []int) []int {
    n := len(nums)
    numCount := make(map[int]int)
    for _, num := range nums {
        numCount[num]++
    }

    miss, dup := 0, 0
    for i := 1; i <= n; i++ {
        if count, ok := numCount[i]; !ok {
            miss = i
        } else if count > 1 {
            dup = i
        }
    }

    return []int{dup, miss}
}
